doc/yaml_config.md

```diff
@@ -0,0 +1,54 @@
+YaML Config
+-----------
+
+Why YaML
+========
+
+With our previous experience we discard at the start XML due to its poor readability.
+JSON format we also consider has serious disadvantage - lack of comments.
+So we decide to go with YaML which is still comfortable to read and also machine read.
+Only disadvantage we found is lack of mature schema support for YaML.
+
+Rules for Config
+================
+
+1. No default values in code. All values have to be defined in yaml config.
+2. No objects, just built-in data types.
+
+Structure
```

Comment:
Maybe would be nice to have a YAML schema or similar, but not sure if there is some solid tool/standard nowadays. Some links that can help:

* https://json-schema-everywhere.github.io/yaml
* https://github.com/dreftymac/awesome-yaml
* https://github.com/datatxt/awesome-yaml

Reply:
well, I found https://asdf-standard.readthedocs.io/en/latest/schemas/yaml_schema.html but it also does not looks like solid/standard.

Reply:
I also feel some kind of schema should be needed. The idea is to have a common structure for:

- The file that holds the configuration of the installer (which is being defined here)
- The options that are passed to the installer via boot parameters
- Some kind of future substitute for the AutoYaST profile

For that we need some kind of schema validation. Maybe not exactly at the yaml level, but we need something.

Reply:
Well, we can also do validation ourself and there are already tools to verify if yaml is valid https://github.com/adrienverge/yamllint .
So some script that verify if everything is defined and with correct types can be also written manually ( but of course some DSL for it would be better ).